### PR TITLE
feat: add environment v2 domain and storage

### PR DESCRIPTION
**Pull request by masaaania**
- This PR adds the EnvironmentV2 domain model & storage implementation.
- API & command implementations suppose to be included in the next PR.

### CODE REVIEW

File: pkg/environment/storage/v2/environment.go [hunk 1]
```diff
@@ -284,3 +293,181 @@ func (s *environmentStorage) ListEnvironments(ctx context.Context,
...
+	query := `
+		SELECT
+			id,
+			name,
+			url_code,
+			description,
+			project_id,
+			archived,
+			created_at,
+			updated_at,
```
Comment (cre8ivejp) [inline pkg/environment/storage/v2/environment.go:380]: Please remove the comma.

File: pkg/environment/storage/v2/environment.go [hunk 2]
```diff
@@ -284,3 +293,181 @@ func (s *environmentStorage) ListEnvironments(ctx context.Context,
...
+	query := fmt.Sprintf(`
+		SELECT
+			id,
+			name,
+			url_code,
+			description,
+			project_id,
+			archived,
+			created_at,
+			updated_at,
```
Comment (cre8ivejp) [inline pkg/environment/storage/v2/environment.go:426]: Please remove the comma.

File: pkg/environment/storage/v2/environment.go [hunk 3]
```diff
@@ -284,3 +293,181 @@ func (s *environmentStorage) ListEnvironments(ctx context.Context,
...
+func (s *environmentStorage) UpdateEnvironmentV2(ctx context.Context, e *domain.EnvironmentV2) error {
+	query := `
+		UPDATE 
+			environment_v2
+		SET
+			name = ?,
+			description = ?,
+			archived = ?,
+			created_at = ?,
+			updated_at = ?,
```
Comment (cre8ivejp) [inline pkg/environment/storage/v2/environment.go:342]: Please remove the comma.

File: pkg/environment/domain/environment.go [hunk 4]
```diff
@@ -54,3 +57,42 @@ func (e *Environment) SetDeleted() {
...
+	*proto.EnvironmentV2
+}
+
+func NewEnvironmentV2(name, urlCode, description, projectID string, logger *zap.Logger) (*EnvironmentV2, error) {
+	id, err := uuid.NewUUID()
+	if err != nil {
+		logger.Error("failed to generate uuid", zap.Error(err))
+		return nil, err
+	}
+	uid := strings.ReplaceAll(id.String(), "-", "")
```
Comment (cre8ivejp) [inline pkg/environment/domain/environment.go:71]: Do we need to remove the hyphen from the id?

Comment (masaaania) [reply pkg/environment/domain/environment.go]: @cre8ivejp 
We don't have to do it 👍 
I fixed!

**Review by cre8ivejp — APPROVED** [review]
Thanks!!

**Pull request merged**